### PR TITLE
CIRCSTORE-496: Keep existing reference/sample data on upgrade

### PR DESCRIPTION
**Pull request by julianladisch**
https://folio-org.atlassian.net/browse/CIRCSTORE-496

Skip installing or resetting reference and sample data if we migrate from a version that already has installed these data.

### CODE REVIEW

File: src/main/java/org/folio/rest/impl/TenantRefAPI.java [hunk 1]
```diff
@@ -41,37 +40,56 @@ Future<Integer> loadData(TenantAttributes attributes, String tenantId,
...
-          .add("circulation-settings-storage/circulation-settings");
+      .compose(r -> loadData(attributes, headers, vertxContext))
+      .compose(r -> registerModule(headers, vertxContext))
+      .mapEmpty();
+  }
+
+  private Future<Integer> loadData(TenantAttributes attributes, Map<String, String> headers,
+      Context vertxContext) {
 
+    log.info("Initializing of tenant's data");
```
Comment (alexanderkurash) [inline src/main/java/org/folio/rest/impl/TenantRefAPI.java:51]: This logging message doesn't follow the established format

File: src/main/java/org/folio/rest/impl/TenantRefAPI.java [hunk 2]
```diff
@@ -41,37 +40,56 @@ Future<Integer> loadData(TenantAttributes attributes, String tenantId,
...
+    TenantLoading tl = new TenantLoading();
```
Comment (alexanderkurash) [inline src/main/java/org/folio/rest/impl/TenantRefAPI.java:52]: tl -> tenantLoading

File: src/main/java/org/folio/rest/impl/TenantRefAPI.java [hunk 3]
```diff
@@ -41,37 +40,56 @@ Future<Integer> loadData(TenantAttributes attributes, String tenantId,
...
-          if (res.failed()) {
-            promise.fail(res.cause());
-          } else {
-            PubSubRegistrationService.registerModule(headers, vertx)
-              .whenComplete((aBoolean, throwable) -> promise.complete());
-          }
-        });
-        return promise.future();
-      });
+  private Future<Boolean> registerModule(Map<String, String> headers, Context vertxContext) {
```
Comment (alexanderkurash) [inline src/main/java/org/folio/rest/impl/TenantRefAPI.java:90]: registerModuleInPubSub

File: src/main/java/org/folio/rest/impl/TenantRefAPI.java [hunk 4]
```diff
@@ -41,37 +40,56 @@ Future<Integer> loadData(TenantAttributes attributes, String tenantId,
...
+      .mapEmpty();
+  }
+
+  private Future<Integer> loadData(TenantAttributes attributes, Map<String, String> headers,
+      Context vertxContext) {
 
+    log.info("Initializing of tenant's data");
+    TenantLoading tl = new TenantLoading();
+    tl.withKey(REFERENCE_KEY).withLead(REFERENCE_LEAD);
+    if (isNew(attributes, "14.0.0")) {
```
Comment (alexanderkurash) [inline src/main/java/org/folio/rest/impl/TenantRefAPI.java:54]: What should we do when we want to change reference data in newer versions? When we update this line ("14.0.0" -> new version), the reference data is going to be updated anyway. However, we'll be in control of that and can let libraries know through Release notes. Was that the intention?

File: src/test/java/org/folio/rest/api/TenantRefApiTests.java [hunk 5]
```diff
@@ -456,6 +461,17 @@ public void migrationShouldNotCorrectFulfillmentPreferenceSpellingFromMigratedVe
       });
   }
 
+  @Test
+  public void keepReferenceData(TestContext context) {
```
Comment (alexanderkurash) [inline src/test/java/org/folio/rest/api/TenantRefApiTests.java:465]: Doesn't it make sense to test all of the reference data types with corresponding module versions, not only Cancellation reasons?

**Review by roman-barannyk — APPROVED** [review]

**Review by OleksandrVidinieiev — APPROVED** [review]

**Pull request merged**